logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt

```diff
@@ -344,114 +314,37 @@ internal class MLSConversationDataSource(
-            }.onSuccess {
-                epochsFlow.emit(groupID)
-            }
-        }
-    }
-
-    private suspend fun sendCommitBundleForExternalCommit(
-        groupID: GroupID,
-        bundle: CommitBundle
-    ): Either<CoreFailure, Unit> =
-        mlsClientProvider.getMLSClient().flatMap { mlsClient ->
-            wrapApiRequest {
-                mlsMessageApi.sendCommitBundle(mlsCommitBundleMapper.toDTO(bundle))
-            }.onFailure {
+        produceAndSendCommitWithRetryAndResult(groupID) {
+            mlsClientProvider.getMLSClient().flatMap { mlsClient ->
                 wrapMLSRequest {
-                    mlsClient.clearPendingGroupExternalCommit(idMapper.toCryptoModel(groupID))
+                    mlsClient.updateKeyingMaterial(idMapper.toCryptoModel(groupID))
                 }
             }.flatMap {
-                wrapMLSRequest {
-                    mlsClient.mergePendingGroupFromExternalCommit(idMapper.toCryptoModel(groupID))
+                wrapStorageRequest {
+                    conversationDAO.updateKeyingMaterial(
+                        idMapper.toCryptoModel(groupID),
+                        DateTimeUtil.currentInstant()
+                    )
                 }
             }
-        }.onSuccess {
-            epochsFlow.emit(groupID)
-        }
-
-    private suspend fun processCommitBundleEvents(events: List<EventContentDTO>) {
-        kaliumLogger.d("Processing commit bundle events")
-        events.forEach { eventContentDTO ->
-            val event =
-                MapperProvider.eventMapper(selfUserId).fromEventContentDTO(
-                    LocalId.generate(),
-                    eventContentDTO
-                )
-            if (event is Event.Conversation) {
-                commitBundleEventReceiver.onEvent(event, EventDeliveryInfo(isTransient = true, source = EventSource.LIVE))
-            }
         }
     }
 
     override suspend fun commitPendingProposals(
         groupID: GroupID
-    ): Either<CoreFailure, Unit> = mutex.withLock {
-        internalCommitPendingProposals(groupID)
-    }
-
-    private suspend fun internalCommitPendingProposals(
-        groupID: GroupID
     ): Either<CoreFailure, Unit> =
-        produceAndSendCommitWithRetry(groupID) {
-            getPendingCommitBundle(groupID)
-        }.flatMap {
-            wrapStorageRequest {
-                conversationDAO.clearProposalTimer(idMapper.toCryptoModel(groupID))
-            }
-        }
-
-    private suspend fun commitPendingProposalsWithoutRetry(
-        groupID: GroupID
-    ): Either<CoreFailure, Unit> =
-        getPendingCommitBundle(groupID).flatMap {
-            if (it != null) {
-                sendCommitBundle(groupID, it)
-            } else {
-                Either.Right(Unit)
-            }
-        }.flatMap {
-            wrapStorageRequest {
-                conversationDAO.clearProposalTimer(idMapper.toCryptoModel(groupID))
+        // TODO Mo here mutex was removed
```

Comment:
@MohamadJaara here mutex was removed

Reply:
since everything is in a transactions it will never be the case that 2 transactions are running at the same time so i guess it is fine